### DISCUSSION TITLE
[@types/node] Wrong return type of `rm` from `node:fs/promises`

### DISCUSSION THREAD

**Discussion by sz3lbi** [DefinitelyTyped/DefinitelyTyped#70153] [Issues with a @types package] [answered]
Hi,
The `fs.rm` function from the `node:fs/promises` package returns the `Promise<void>` type, while the docs say that it should be `Promise<undefined>` on success:
> Returns: `<Promise>` Fulfills with `undefined` upon success.

(https://nodejs.org/api/fs.html#fspromisesrmpath-options)

Node v22.5.1

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment, marked as answer]
Actually nothing passed to `resolve` function - https://github.com/nodejs/node/blob/6ad25e37b570124d82e6cb25e237592c9b927ec4/lib/internal/fs/rimraf.js#L162

In this case the promise gets fulfilled with `undefined` as its value. That is why it documented as `Fulfills with undefined upon success.`.

In TypeScript we have special type `void` for cases when function doesn't return any explicit value or promise doesn't;t fulfilled with any explicit value.

@sz3lbi are you really need this `undefined` value? Could you provide code examples where you need it?

**Reply by sz3lbi** [reply]
You're right that it's not necessary in my case. Thanks for the clarification.
